articles/112_generated_interfaces_cpp.md

```diff
@@ -28,16 +30,25 @@ This article specifies the generated C++ code for ROS interface types defined in
 ## Namespacing
 
 All code of a ROS package should be defined in a namespace named after the package.
-To separate the generated code from other code within the package it is defined in a sub namespace:
+To separate the generated code from other code within the package it is defined in a sub-namespace:
+
 * namespace for ROS messages: `package_name::msg`.
 * namespace for ROS services: `package_name::srv`.
 
+<div class="alert alert-warning" markdown="1">
+  **Note:** Those namespaces are not consistent with ROS 1 usage.
+</div>
+
 
 ## Generated files
 
 Following the C++ style guide of ROS 2 the namespace hierarchy is mapped to a folder structure.
 The filenames use lowercase alphanumeric characters with underscores for separating words and end with either `.hpp` or `.cpp`.
 
+<div class="alert alert-warning" markdown="1">
+  **Note:** The suggested naming is not consistent with ROS 1 usage.
+</div>
+
```

Comment:
> With your proposed change how can ROS 1 and ROS 2 header be included at the same time?

Through some other ugly hack that only affects those few programs. The first one that comes to mind is playing around with the include paths returned by pkg_config for the ROS1 tree. Yep, that's ugly.

If you don't come up with a better one before I get back from vacation, I promise to come up with several other ideas you probably won't like, either. :smile: 

> How will the C equivalent of the message header be named then?

However you like. The scope of this document is C++ only. There is little or no existing ROS code in C, so you are free to invent new solutions.

Personally, I would investigate making the C++ headers bilinugal for both C and C++. But, maybe that's not feasible. It's up to you, Dirk.